lib/services/offered_food_service.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:zachranobed/models/offered_food.dart';
+import 'package:zachranobed/services/helper_service.dart';
+
+class OfferedFoodService {
+  final _offeredFoodCollection =
+      FirebaseFirestore.instance.collection('nabidka').withConverter(
+    fromFirestore: (snapshot, options) {
+      final json = snapshot.data() ?? {};
+      json['id'] = snapshot.id;
+      return OfferedFood.fromJson(json);
+    },
+    toFirestore: (value, options) {
+      final json = value.toJson();
+      json.remove('id');
+      return json;
+    },
+  );
+
+  Stream<List<OfferedFood>> loggedUserOfferedFoodStream(
+      {required BuildContext context,
+      int? limit,
+      String? additionalFilterField,
+      dynamic additionalFilterValue}) {
+    var query = _offeredFoodCollection.orderBy('date', descending: true);
+
+    if (limit != null) {
+      query = query.limit(limit);
+    }
+
+    if (additionalFilterField != null && additionalFilterValue != null) {
+      query =
+          query.where(additionalFilterField, isEqualTo: additionalFilterValue);
+    }
+
+    return query.snapshots().map((querySnapshot) => querySnapshot.docs
+        .map((docSnapshot) => docSnapshot.data())
+        .where((offeredFood) =>
+            offeredFood.donor ==
+            HelperService.getCurrentUser(context)!.establishmentName)
+        .toList());
+  }
+
+  // TODO - z tohodle by bylo fajn nějak udělat stream
```

Comment:
Jen se to musí vyhnout tomu aby se vždy četlo x dokumentů. To by kreditka rychle zmizla 😄 Můžeme pak někdy probrat na callu, dá se to shraňovat i jinak